A50-xds-outlier-detection.md

```diff
@@ -186,7 +186,74 @@ The subchannel wrapper will track the latest state update from the underlying su
+The `xds_cluster_resolver` config described in [gRFC A37: xDS Aggregate and Logical DNS Clusters](https://github.com/grpc/proposal/blob/master/A37-xds-aggregate-and-logical-dns-clusters.md) will be modified so that the `DiscoveryMechanism` inner message contains an additional `outlier_detection` field as follows:
+
+```proto
+message XdsClusterResolverLoadBalancingPolicyConfig {
+  // Describes a discovery mechanism instance.
+  // For EDS or LOGICAL_DNS clusters, there will be exactly one
+  // DiscoveryMechanism, which will describe the cluster of the parent
+  // CDS policy.
+  // For aggregate clusters, there will be one DiscoveryMechanism for each
+  // underlying cluster.
+  message DiscoveryMechanism {
+    // Cluster name.
+    string cluster = 1;
+
+    // LRS server to send load reports to.
+    // If not present, load reporting will be disabled.
+    // If set to the empty string, load reporting will be sent to the same
+    // server that we obtained CDS data from.
+    google.protobuf.StringValue lrs_load_reporting_server_name = 2;
+
+    // Maximum number of outstanding requests can be made to the upstream cluster.
+    // Default is 1024.
+    google.protobuf.UInt32Value max_concurrent_requests = 3;
+
+    enum Type { EDS, LOGICAL_DNS };
+    Type type = 4;
+
+    // For type EDS only.
+    // EDS service name, as returned in CDS.
+    // May be unset if not specified in CDS.
+    string eds_service_name = 5;
+
+    // For type LOGICAL_DNS only.
+    // DNS name to resolve in "host:port" form.
+    string dns_hostname = 6;
+
+    // The configuration for outlier_detection child policies
+    // Within this message, the child_policy field will be ignored
+    OutlierDetectionLoadBalancingConfig outlier_detection = 7;
+  }
+
+  // Ordered list of discovery mechanisms.
+  // Must have at least one element.
+  // Results from each discovery mechanism are concatenated together in successive
+  // priorities.
+  repeated DiscoveryMechanism discovery_mechanisms = 1;
+
+  // Locality-picking policy.
+  // This policy's config is expected to be in the format used
+  // by the weighted_target policy.  Note that the config should include
+  // an empty value for the "targets" field; that empty value will be
+  // replaced by one that is dynamically generated based on the EDS data.
+  // Optional; defaults to "weighted_target".
+  repeated LoadBalancingConfig locality_picking_policy = 2;
```

Comment:
These two fields are actually no longer used; this was updated by [gRFC A42](https://github.com/grpc/proposal/blob/master/A42-xds-ring-hash-lb-policy.md#change-child-policy-config-generation-in-xds_cluster_resolver-policy).

I think we should probably just bypass this kind of issue by listing only the new field to be added, not showing any of the additional existing fields.

As a follow-up, you can put together a PR for the grpc/grpc-proto repo to make the changes to the service config protos.

Reply:
Fixed. The follow-up PR is at grpc/grpc-proto#104.